web-server/v0.4/src/models/search.js

```diff
@@ -53,11 +53,12 @@ export default {
         let searchResults = {};
         searchResults['resultCount'] = response.hits.total;
         let parsedResults = [];
-        response.hits.hits.map(result => {
+        response.hits.hits.map((result, index) => {
```

Comment:
If `response.hits.hits` does not have a `result` and an `index` field, what happens here?

Reply:
The `map()` method takes the array of hits and will always return the value which is designated as `result` in this case in addition to the array `index` currently being iterated on. If `response.hits.hits` is an empty array, the `map()` method would be skipped and as a result the reducer would contain its default value for search results. This would then render an empty table in the search view showing a "No data found" message in the table which could mean the user had a bad query.

Reply:
Can you open an issue to check for errors when response has no `hits`, and/or when it does not have a  `result` or `index` field?

Reply:
I've opened issue #1167 to track this behavior. Ideally, this would encompass all cases where we would potentially continue with post processing after receiving either empty data or a network error.